tika-parsers/tika-parsers-standard/tika-parsers-standard-modules/tika-parser-cad-module/src/main/java/org/apache/tika/parser/dwg/AbstractDWGParser.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.tika.parser.dwg;
+
+
+import org.apache.tika.config.Field;
+import org.apache.tika.parser.AbstractParser;
+import org.apache.tika.parser.ParseContext;
+
+
+
+
+public abstract class AbstractDWGParser extends AbstractParser {
+
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 6261810259683381984L;
+	private final DWGParserConfig defaultDwgParserConfig = new DWGParserConfig();
+
+    public void configure(ParseContext parseContext) {
+    	DWGParserConfig dwgParserConfig =  parseContext.get(DWGParserConfig.class, defaultDwgParserConfig);
+        parseContext.set(DWGParserConfig.class, dwgParserConfig);
+    }
+
+
+    String getDwgReadExecutable() {
+        return defaultDwgParserConfig.getDwgReadExecutable();
+    }
+    
+    @Field
+    public void setDwgReadExecutable(String dwgReadExecutable) {
+    	defaultDwgParserConfig.setDwgReadExecutable(dwgReadExecutable);
+    }
+    
+    boolean isCleanDwgReadOutput() {
+        return defaultDwgParserConfig.isCleanDwgReadOutput();
+    }
+    
+    @Field
+    public void setCleanDwgReadOutput(boolean cleanDwgReadOutput) {
+    	defaultDwgParserConfig.setCleanDwgReadOutput(cleanDwgReadOutput);
+    }
+    
+    int getCleanDwgReadOutputBatchSize() {
+        return defaultDwgParserConfig.getCleanDwgReadOutputBatchSize();
+    }
+    
+    @Field
+    public void setCleanDwgReadOutputBatchSize(int cleanDwgReadOutputBatchSize) {
+    	defaultDwgParserConfig.setCleanDwgReadOutputBatchSize(cleanDwgReadOutputBatchSize);
+    }
+    String getCleanDwgReadRegexToReplace() {
+        return defaultDwgParserConfig.getCleanDwgReadRegexToReplace();
+    }
+    
+    @Field
+    public void setCleanDwgReadRegexToReplace(String cleanDwgReadRegexToReplace) {
+    	defaultDwgParserConfig.setCleanDwgReadRegexToReplace(cleanDwgReadRegexToReplace);
+    }
+    String getCleanDwgReadReplaceWith() {
+        return defaultDwgParserConfig.getCleanDwgReadReplaceWith();
+    }
+    
+    @Field
+    public void setCleanDwgReadReplaceWith(String cleanDwgReadReplaceWith) {
+    	defaultDwgParserConfig.setCleanDwgReadReplaceWith(cleanDwgReadReplaceWith);
+    }
+    
```

Comment:
May want to implement Initializable and see if the dwg executable is actually working.  Please add lots of logging if a user specified a dwg executable, but it isn't working.

Something along the lines of: https://github.com/apache/tika/blob/main/tika-parsers/tika-parsers-standard/tika-parsers-standard-modules/tika-parser-ocr-module/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java#L529

Reply:
If a user doesn't have dwgread on their system, will this throw a config exception during initialization?

Reply:
> 

Hi @tballison 
It seems to work ok, I've compiled a version of tika server and if I don't supply the parser config it falls back to the old parser. If the config is incorrect then Tika server fails to start (which is what we want?)

Parser config looks like this:
```
<parser class="org.apache.tika.parser.dwg.DWGReadParser">
	<params>
		<param name="dwgReadExecutable" type="string">G:/libredwg-0.12.5-win64/dwgread.exe</param>
		<param name="cleanDwgReadOutput" type="bool">true</param>
		<param name="cleanDwgReadOutputBatchSize" type="int">10000000</param>
		<param name="dwgReadTimeout" type="long">10</param>
		<param name="cleanDwgReadRegexToReplace" type="string">[^\x20-\x7e]| nan,| nan </param>
		<param name="cleanDwgReadReplaceWith" type="string"/>
	</params>
</parser>

Reply:
W00t!